### ISSUE TITLE
ARM: container linkerd-proxy is not ready

### ISSUE THREAD

**Issue by ctrahey** [linkerd/linkerd2#5626]
### What is the issue?

After a fresh install with no customizations (same applies to the cluster), linkerd-proxy containers are failing readiness checks with 503. 

### How can it be reproduced?
Kubernetes 1.20 on Raspberry Pi model 4b 4GB RAM. (my cluster is 3 control plane + 6 workers)
Calico CNI
Follow Linkerd [install instructions](https://linkerd.io/2/getting-started/)
Install proceeds through pod scheduling, `linkerd check` ends with:

`pod/linkerd-controller-864657567f-pp5ml container linkerd-proxy is not ready`

### Logs, error output, etc
Most interesting log output is from proxy:
`Readiness probe failed: HTTP probe failed with statuscode: 503`
[Logs.gist](https://gist.github.com/ctrahey/797ce0707f3bc658f3b5e37a8096b767)

#### `linkerd check` output

```text
15-MBP:~ christophertrahey$ linkerd check --proxy
kubernetes-api
--------------
√ can initialize the client
√ can query the Kubernetes API

kubernetes-version
------------------
√ is running the minimum Kubernetes API version
√ is running the minimum kubectl version

linkerd-existence
-----------------
√ 'linkerd-config' config map exists
√ heartbeat ServiceAccount exist
√ control plane replica sets are ready
√ no unschedulable pods
√ controller pod is running
√ can initialize the client
√ can query the control plane API

linkerd-config
--------------
√ control plane Namespace exists
√ control plane ClusterRoles exist
√ control plane ClusterRoleBindings exist
√ control plane ServiceAccounts exist
W0127 19:11:09.883635   81044 warnings.go:67] apiextensions.k8s.io/v1beta1 CustomResourceDefinition is deprecated in v1.16+, unavailable in v1.22+; use apiextensions.k8s.io/v1 CustomResourceDefinition
√ control plane CustomResourceDefinitions exist
W0127 19:11:09.898357   81044 warnings.go:67] admissionregistration.k8s.io/v1beta1 MutatingWebhookConfiguration is deprecated in v1.16+, unavailable in v1.22+; use admissionregistration.k8s.io/v1 MutatingWebhookConfiguration
√ control plane MutatingWebhookConfigurations exist
W0127 19:11:09.910050   81044 warnings.go:67] admissionregistration.k8s.io/v1beta1 ValidatingWebhookConfiguration is deprecated in v1.16+, unavailable in v1.22+; use admissionregistration.k8s.io/v1 ValidatingWebhookConfiguration
√ control plane ValidatingWebhookConfigurations exist
√ control plane PodSecurityPolicies exist

linkerd-identity
----------------
√ certificate config is valid
√ trust anchors are using supported crypto algorithm
√ trust anchors are within their validity period
√ trust anchors are valid for at least 60 days
√ issuer cert is using supported crypto algorithm
√ issuer cert is within its validity period
√ issuer cert is valid for at least 60 days
√ issuer cert is issued by the trust anchor

linkerd-webhooks-and-apisvc-tls
-------------------------------
√ tap API server has valid cert
√ tap API server cert is valid for at least 60 days
W0127 19:11:10.008082   81044 warnings.go:67] admissionregistration.k8s.io/v1beta1 MutatingWebhookConfiguration is deprecated in v1.16+, unavailable in v1.22+; use admissionregistration.k8s.io/v1 MutatingWebhookConfiguration
√ proxy-injector webhook has valid cert
√ proxy-injector cert is valid for at least 60 days
W0127 19:11:10.039035   81044 warnings.go:67] admissionregistration.k8s.io/v1beta1 ValidatingWebhookConfiguration is deprecated in v1.16+, unavailable in v1.22+; use admissionregistration.k8s.io/v1 ValidatingWebhookConfiguration
√ sp-validator webhook has valid cert
√ sp-validator cert is valid for at least 60 days

linkerd-identity-data-plane
---------------------------
√ data plane proxies certificate match CA

linkerd-api
-----------
- pod/linkerd-controller-864657567f-pp5ml container linkerd-proxy is not ready E0127 19:11:29.941541   81044 portforward.go:233] lost connection to pod
× control plane pods are ready
    pod/linkerd-controller-864657567f-pp5ml container linkerd-proxy is not ready
    see https://linkerd.io/checks/#l5d-api-control-ready for hints
```

### Environment

- Kubernetes Version: 1.20
- Cluster Environment: Bare Metal Raspberry Pi 4b, 4GB
- Host OS: Ubuntu 20.04
- Linkerd version: 2.9.2

### Possible solution
I've read [this issue](https://github.com/linkerd/linkerd2/discussions/4685) and a couple others with similar error messages but nothing jumped out at me for this case. I've read some things about TLS issues, but I don't know how to troubleshoot, and I don't believe I am doing anything unique there.

**Comment by ctrahey** [comment]
Optimistically closing this while I troubleshoot - I just realized my calico pods are not healthy.

**Comment by ctrahey** [comment]
Re-opening, I fixed Calico but my issue remains.

**Comment by Pothulapati** [comment]
@ctrahey We do have some integration tests that test with Calico and Kind and different integration tests that test out in ARM. Can you increase the log-level and see if there is anything weird?

**Comment by ctrahey** [comment]
Thanks Tarun,
I must apologize for the noise here - I needed to reboot my control plane nodes, and I'm not even sure that's what fixed it - but by the time I ran linkerd check again everything was up. Previously, I waited overnight and it was still failing.

I uninstalled linkerd, then re-installed and everything came up in a timely fashion.
I'll mention this on the off chance this is related: 
The change I made to control plane nodes was extending haproxy timeout (on the kube api endpoint). It had been 20s and I was getting annoyed at my streaming log commands dropping out. Not sure if linkerd needs to do something with the kube api that takes longer than 20s...?
